leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/impl/RegisterResource.java

```diff
@@ -254,9 +254,9 @@ private Identity extractIdentity(CoapExchange exchange) {
                 return Identity.rpk(peerAddress, publicKey);
             } else if (senderIdentity instanceof X500Principal) {
                 // Extract common name
-                Matcher endpointMatcher = Pattern.compile("CN=.*?,").matcher(senderIdentity.getName());
+                Matcher endpointMatcher = Pattern.compile("CN=(.*?)(,|$)").matcher(senderIdentity.getName());
                 if (endpointMatcher.find()) {
-                    String x509CommonName = endpointMatcher.group().substring(3, endpointMatcher.group().length() - 1);
+                    String x509CommonName = endpointMatcher.group(1);
```

Comment:
We can just move an utility method in `leshan-core`, but maybe it's time to ask again for a `leshan-core-cf` project.

We didn't create it at the beginning because we didn't have so much common code. After a quick look, I think we have 6,7 classes which could go in a `leshan-core-cf` package.

Should we create a new  `leshan-core-cf` project ?